File: src/UI/src/features/employees/types/employee.ts
import { z } from 'zod';
import { employeeDepartmentSchema } from '@features/employees/types/employeeDepartment.ts';

export const employeeSchema = z.object({
  businessEntityID: z.number(),
  nationalIDNumber: z.string(),
  loginID: z.string(),
  firstName: z.string(),
  middleName: z.string(),
  lastName: z.string(),
  jobTitle: z.string(),
  birthDate: z.preprocess((arg) => {
    if (typeof arg === 'string' || arg instanceof Date) return new Date(arg);
  }, z.date()),
  maritalStatus: z.enum(['Single', 'Married']),
  gender: z.enum(['Male', 'Female']),
  hireDate: z.preprocess((arg) => {
    if (typeof arg === 'string' || arg instanceof Date) return new Date(arg);
  }, z.date()),
  salariedFlag: z.boolean(),
  vacationHours: z.number(),
  sickLeaveHours: z.number(),
  currentFlag: z.boolean(),
  department: employeeDepartmentSchema
});

export type Employee = z.infer<typeof employeeSchema>;

File: src/API/AdventureWorks.Domain/Employees/Employee.cs
namespace AdventureWorks.Domain.Employees;

public class Employee
{
    public required int BusinessEntityID { get; init; }
    public required string NationalIDNumber { get; init; }
    public required string LoginID { get; init; }
    public required string FirstName { get; init; }
    public required string MiddleName { get; init; }
    public required string LastName { get; init; }
    public required string JobTitle { get; init; }
    public required DateOnly BirthDate { get; init; }
    public required MartialStatus MaritalStatus { get; init; }
    public required Gender Gender { get; init; }
    public required DateOnly HireDate { get; init; }
    public required bool SalariedFlag { get; init; }
    public required double VacationHours { get; init; }
    public required double SickLeaveHours { get; init; }
    public required bool CurrentFlag { get; init; }
    public required Department Department { get; init; }
}